server/api/attendance/attendance.controller.js

```diff
@@ -613,6 +613,45 @@ exports.setAttendanceFullBonus = function(req, res) {
 };
 // *******************************************************
 
+// *******************************************************
+// Adds an attendance entry with the given parameters
+//
+// Restricted to admins
+// router.post('/attend/:user/manual', auth.hasRole('admin'), controller.attend);
+exports.addManualAttendance = function(req, res) {
+  var userId = req.params.user;
+  var date = req.body.date;
+  var smallgroup = req.body.smallgroup
+  var bonusDay = req.body.bonusday;console.log("1")
+
+  return User.findById(userId, function(err,user){
+    if (err) {
+      return handleError(err);
+    }console.log("2")
+    return ClassYear.getCurrent(function(err, classYear){
+      if (err) {
+        return handleError(err);
+      }console.log("3")
+
```

Comment:
You should first check that there isn't already an attendance of that type for the user submitted on that day.